tracer/span.go

```diff
@@ -57,223 +53,171 @@ type Span struct {
-
+// SetBaggageItem sets a key/value pair as baggage on the span.
+func (s *span) SetBaggageItem(key, val string) dd.Span {
 	s.Lock()
 	defer s.Unlock()
 
-	s.setMeta(key, value)
-
+	s.context.setBaggageItem(key, val)
+	return s
 }
 
-// SetMetas adds arbitrary meta fields from a given map to the current Span.
-// If the Span has been finished, it will not be modified by the method.
-func (s *Span) SetMetas(metas map[string]string) {
-	for k, v := range metas {
-		s.SetMeta(k, v)
-	}
-}
-
-// GetMeta will return the value for the given tag or the empty string if it
-// doesn't exist.
-func (s *Span) GetMeta(key string) string {
-	if s == nil {
-		return ""
-	}
-	s.RLock()
-	defer s.RUnlock()
-	if s.Meta == nil {
-		return ""
-	}
-	return s.Meta[key]
-}
-
-// SetMetrics adds a metric field to the current Span.
-// DEPRECATED: Use SetMetric
-func (s *Span) SetMetrics(key string, value float64) {
-	if s == nil {
-		return
-	}
-	s.SetMetric(key, value)
-}
-
-// SetMetric sets a float64 value for the given key. It acts
-// like `set_meta()` and it simply add a tag without further processing.
-// This method doesn't create a Datadog metric.
-func (s *Span) SetMetric(key string, val float64) {
-	if s == nil {
-		return
-	}
-
+// BaggageItem gets the value for a baggage item given its key. Returns the
+// empty string if the value isn't found in this Span.
+func (s *span) BaggageItem(key string) string {
 	s.Lock()
 	defer s.Unlock()
 
-	// We don't lock spans when flushing, so we could have a data race when
-	// modifying a span as it's being flushed. This protects us against that
-	// race, since spans are marked `finished` before we flush them.
-	if s.finished {
-		return
-	}
-
-	if s.Metrics == nil {
-		s.Metrics = make(map[string]float64)
-	}
-	s.Metrics[key] = val
+	return s.context.baggage[key]
 }
 
-// SetError stores an error object within the span meta. The Error status is
-// updated and the error.Error() string is included with a default meta key.
-// If the Span has been finished, it will not be modified by this method.
-func (s *Span) SetError(err error) {
-	if err == nil || s == nil {
-		return
-	}
+const (
+	// spanType defines the Span type (web, db, cache)
+	spanType = "span.type"
+	// serviceName defines the Service name for this Span
+	serviceName = "service.name"
+	// resourceName defines the Resource name for the Span
+	resourceName = "resource.name"
+)
 
+// SetTag adds a tag to the span, overwriting pre-existing values for
+// the given key.
+func (s *span) SetTag(key string, value interface{}) dd.Span {
 	s.Lock()
 	defer s.Unlock()
 	// We don't lock spans when flushing, so we could have a data race when
 	// modifying a span as it's being flushed. This protects us against that
 	// race, since spans are marked `finished` before we flush them.
 	if s.finished {
-		return
+		return s
+	}
+	if v, ok := toFloat64(value); ok {
+		// sent as numeric value, so we can store it as a metric
+		switch key {
+		case "sampling.priority":
```

Comment:
we may convert this one to a constant so that we can change it (if needed) without introducing a breaking change